x-pack/plugins/apm/public/components/routing/templates/apm_service_template/analyze_data_button.test.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { composeStories } from '@storybook/testing-react';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import {
+  ENVIRONMENT_ALL,
+  ENVIRONMENT_NOT_DEFINED,
+} from '../../../../../common/environment_filter_values';
+import * as stories from './analyze_data_button.stories';
+
+const { Example } = composeStories(stories);
+
+describe('AnalyzeDataButton', () => {
+  describe('with a non-RUM and non-mobile agent', () => {
+    it('renders nothing', () => {
+      render(<Example agentName="java" />);
+
+      expect(screen.queryByRole('link')).not.toBeInTheDocument();
+    });
+  });
+
+  describe('with a RUM agent', () => {
+    it('uses a ux dataType', () => {
+      render(<Example agentName="rum-js" />);
+
+      expect((screen.getByRole('link') as HTMLAnchorElement).href).toEqual(
+        'http://localhost/app/observability/exploratory-view#?sr=(apm-series:(dt:ux,isNew:!t,op:average,rdf:(service.environment:!(testEnvironment),service.name:!(testServiceName)),rt:kpi-over-time,time:(from:now-15m,to:now)))'
+      );
+    });
+  });
+
+  describe('with a mobile agent', () => {
+    it('uses a mobile dataType', () => {
+      render(<Example agentName="iOS/swift" />);
+
+      expect((screen.getByRole('link') as HTMLAnchorElement).href).toEqual(
+        'http://localhost/app/observability/exploratory-view#?sr=(apm-series:(dt:mobile,isNew:!t,op:average,rdf:(service.environment:!(testEnvironment),service.name:!(testServiceName)),rt:kpi-over-time,time:(from:now-15m,to:now)))'
+      );
+    });
+  });
+
+  describe('with no environment', () => {
+    it('does not include the environment', () => {
+      render(<Example environment={undefined} />);
+
+      expect((screen.getByRole('link') as HTMLAnchorElement).href).toEqual(
+        'http://localhost/app/observability/exploratory-view#?sr=(apm-series:(dt:mobile,isNew:!t,op:average,rdf:(service.name:!(testServiceName)),rt:kpi-over-time,time:(from:now-15m,to:now)))'
+      );
+    });
+  });
+
+  describe('with environment not defined', () => {
+    it('does not include the environment', () => {
+      render(<Example environment={ENVIRONMENT_NOT_DEFINED.value} />);
+
+      expect((screen.getByRole('link') as HTMLAnchorElement).href).toEqual(
+        'http://localhost/app/observability/exploratory-view#?sr=(apm-series:(dt:mobile,isNew:!t,op:average,rdf:(service.name:!(testServiceName)),rt:kpi-over-time,time:(from:now-15m,to:now)))'
```

Comment:
I'm a little confused as to why this will result in the same link as ENVIRONMENT_ALL. Should this include a `not (service.environment:*)` filter?

Reply:
@shahzad31 any reason why these query parameters are abbreviated?

Reply:
It shouldn't be the same link. I'm pushing up a change to make ENVIRONMENT_ALL use ALL_VALUES and leave it blank for not defined.

Reply:
Hmm, I still don't see how ENVIRONMENT_NOT_DEFINED is properly propagated to the Exploratory View. Is it even possible with the current DSL?

Reply:
Yes there might be some limitation, Essentially ALL_VALUES doesn't means it adds term filter with :* , it just doesn't adds a filter in the query. 

Though it makes a difference in the UI, that combox box is filled with ALL_VALUES pill.

Regarding abbreviation, that was added just out of fair of not exceeding url limit. We are hoping to move it to session storage. At that point , it will still be possible to generate url. But as soon as user is in exp view, it will be moved to session storage. I guess it may make sense to get rid of abbreviation since it increases the readability.

Reply:
`ENVIRONMENT_NOT_DEFINED` if this generates the exists filter in es query, then yes it's not possible with current implementation.

Reply:
It seems like it would be OK for now to just bring you there with the filter blank, like this is doing. I can open an issue to handle ENVIRONMENT_NOT_DEFINED at some point.

Reply:
Created #105989 to address ENVIRONMENT_NOT_DEFINED.